lightning/src/ln/msgs.rs

```diff
@@ -1512,22 +1510,20 @@ impl Writeable for UnsignedChannelUpdate {
 
 impl Readable for UnsignedChannelUpdate {
 	fn read<R: Read>(r: &mut R) -> Result<Self, DecodeError> {
-		let has_htlc_maximum_msat;
 		Ok(Self {
 			chain_hash: Readable::read(r)?,
 			short_channel_id: Readable::read(r)?,
 			timestamp: Readable::read(r)?,
 			flags: {
 				let flags: u16 = Readable::read(r)?;
-				let message_flags = flags >> 8;
-				has_htlc_maximum_msat = (message_flags as i32 & 1) == 1;
+				// Note: we ignore `must_be_one`, formely `message_flags`, since it was deprecated by the spec.
 				flags as u8
 			},
 			cltv_expiry_delta: Readable::read(r)?,
 			htlc_minimum_msat: Readable::read(r)?,
 			fee_base_msat: Readable::read(r)?,
 			fee_proportional_millionths: Readable::read(r)?,
-			htlc_maximum_msat: if has_htlc_maximum_msat { Readable::read(r)? } else { OptionalField::Absent },
+			htlc_maximum_msat: Readable::read(r)?,
```

Comment:
> Hmmmm, so I think missing the field will cause us to disconnect our peer because they sent us a bogus message we couldn't read. I think maybe the right fix is to change how we handle `wire::read` failures in `PeerManager` to ignore errors on gossip messages. 

AFAICT, we would just send a warning and continue message processing (cf. [this line ff.](https://github.com/lightningdevkit/rust-lightning/blob/main/lightning/src/ln/peer_handler.rs#L1020))?  But maybe I'm looking in the wrong place?

> That would also let us drop the stupid `Result<PublicKey, secp256k1::Error>` crap in some of the gossip message fields.

Also not quite sure what you're referring to here, the only place I can find a `Result<PublicKey, secp256k1::Error>` signature in `msgs::OnionPacket`?